### PR TITLE
feat: [#56] add test for full coverage

### LINKED ISSUE #56: Expand backend integration tests
Context: https://github.com/cyfronet-fid/backoffice-ordering-system/pull/54 introduced a stub/base for implementing integration tests (tests that do not mock anything and hit an actual database)

Expand on all scenarios (happy paths, failures, edge cases) in the corresponding `tests/integrations/routers/*` so that we have coverage. Pay special attention to correct RBAC behavior

Fix bugs if the tests catch any unexpected behaviors.

Notes: 
- If you run tests via `poetry run pytest` you should be getting test coverage reported. We should aim at +/- 100% coverage for the routers.
- you can add a coverage flag to README (https://github.com/marketplace/actions/code-coverage-annotations)

### PR DESCRIPTION
**Pull request by sqrasdf**
closes #56

### CODE REVIEW

**Review by wujuu — CHANGES_REQUESTED** [review]
Good progress on test coverage — nice work on the RBAC tests for orders/providers/users.

Main themes:

1. **Cleanup**: remove `print` statements (test_api.py:185-186), commented-out code (api.py:101), and unused imports (test_call_whitelabel.py:4)
2. **One assertion per test**: split multi-scenario tests (`test_api_requires_valid_key`, `test_create_order_fails_when_owner_missing_or_not_mp_user`)
3. **Use existing fixtures**: many tests manually do `app.dependency_overrides[current_user]` + try/finally — this is what `mp_user_client` / `provider_manager_client` already provide. Applies across test_orders, test_providers, and test_users.
4. **Broken assertion**: `test_get_order_messages_sorted_by_created_at` doesn't actually verify ordering (see inline comment)
5. **Missing coverage**: GET message endpoints need RBAC scoping tests; `get_whitelabel_role` missing MP_USER case
6. **Production code in test PR**: the `api.py` Message construction change should be a separate PR or at least cleaned up

File: backend/tests/integration/routers/test_providers.py [hunk 1]
```diff
@@ -1,10 +1,85 @@
...
+):
+    provider = provider_factory()
+    pm = provider_manager_user_factory()
+    provider.managers.append(pm)
+    db_session.commit()
+
+    mp_user = User(name="mp", email="mp@example.com", user_type=[UserType.MP_USER])
+    db_session.add(mp_user)
+    db_session.commit()
+
```
Comment (wujuu) [inline backend/tests/integration/routers/test_providers.py]: this `User` is leftover from before the refactor — `mp_user_client` already has its own user, so lines 39-42 just create a dangling row in the DB. can be removed. that also makes the `User` and `UserType` imports at the top unused

File: backend/tests/integration/routers/test_orders.py [hunk 2]
```diff
@@ -1,10 +1,129 @@
-def test_admin_sees_all_orders(admin_client, seeded_order):
+from http import HTTPStatus
+
+from backend.models.tables import OrderStatus, UserType
+from tests.integration.routers.conftest import provider_manager_client
```
Comment (wujuu) [inline backend/tests/integration/routers/test_orders.py]: pytest fixtures are auto-discovered from `conftest.py` — you don't need to import them explicitly. same thing in `test_providers.py:4` and `test_users.py:4`. just use them as function params and pytest resolves them

also `UserType` on line 3 is no longer used in this file
